internal/fleet/lifecycle.go

```
@@ -0,0 +1,259 @@
+		return Profile{}, false, fmt.Errorf("AGX-FLEET-PROFILE-READ: cannot inspect Deployment Profile: %w", err)
+	}
+	if err := metadatafile.RequireRealEntry(path, info, false, "Deployment Profile", "AGX-FLEET-PROFILE-INVALID"); err != nil {
+		return Profile{}, false, err
+	}
+	file, err := os.Open(path)
+	if err != nil {
+		return Profile{}, false, fmt.Errorf("AGX-FLEET-PROFILE-READ: cannot open Deployment Profile: %w", err)
+	}
+	openedInfo, statErr := file.Stat()
+	if statErr != nil || !openedInfo.Mode().IsRegular() || !os.SameFile(info, openedInfo) {
+		file.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `file.Close` is not checked

(errcheck)